src/main/scala/io/scalac/tezos/translator/service/LibraryService.scala

```diff
@@ -49,6 +49,24 @@ class LibraryService(repository: LibraryRepository, log: LoggingAdapter)(implici
     } yield updated
   }
 
+  def getRecord(id: LibraryEntryId): Future[LibraryEntry] = {
+    val uidNotExistsException =
+      Future.failed(new IllegalArgumentException(s"Library Entry does not exist for uid: $id"))
+
+    for {
+      entry <- repository.get(id).flatMap {
```

Comment:
this should return only accepted entries !

Reply:
done but need refactoring